GSA_Adapter/CRUD/Create.cs

```diff
@@ -91,22 +94,23 @@ private bool CreateLinks(IEnumerable<RigidLink> links)
             bool success = true;
             foreach (RigidLink link in links)
             {
-                success &= ComCall(Convert.ToGsaString(link, link.CustomData[AdapterIdName].ToString(), 0));
+                success &= ComCall(Convert.ToGsaString(link, link.GSAId().ToString(), 0));
             }
 
             foreach (RigidLink link in links)
             {
-                List<string> allIds = new List<string>();
+                List<int> allIds = new List<int>();
                 for (int i = 1; i < link.SecondaryNodes.Count; i++)
                 {
-                    string id =  NextFreeId(link.GetType(), i == 1).ToString();
-                    success &= ComCall(Convert.ToGsaString(link, id, i));
+                    int id =  (int)NextFreeId(link.GetType(), i == 1);
+                    success &= ComCall(Convert.ToGsaString(link, id.ToString(), i));
                     allIds.Add(id);
                 }
                 if (link.SecondaryNodes.Count > 1)
                 {
-                    allIds.Add(link.CustomData[AdapterIdName].ToString());
-                    link.CustomData[AdapterIdName + "-AllIds"] = allIds;
+                    link.Fragments.Remove(typeof(GSAId)); // to remove the existing single id on the link
+                    allIds.Add(link.AdapterId<int>(typeof(GSAId)));
```

Comment:
Order switch here of these two lines I think is critical
```suggestion
                    allIds.Add(GetAdapterId<int>(link));
                    link.Fragments.Remove(typeof(GSAId)); // to remove the existing single id on the link
```